### DISCUSSION TITLE
does deleting db rows frees up space?

### DISCUSSION THREAD

**Discussion by Nisthar** [supabase/supabase#1467] [Questions] [answered]
Supabase database only gives you 500mb storage for free tier. Right? i am wondering does deleting rows periodically frees up this space or does the deleted rows count too?

**Comment by silentworks** [comment, marked as answer]
@Nisthar if the row is deleted it should free up space as it's actually deleted from the disk.

**Reply by steve-chavez** [reply]
> as it's actually deleted from the disk.

@silentworks Ah, that will not actually happen until a VACUUM(postgres garbage collector) is run. From [postgres docs](https://www.postgresql.org/docs/current/sql-vacuum.html):

> In normal PostgreSQL operation, tuples that are deleted or obsoleted by an update are not physically removed from their table; they remain present until a VACUUM is done. 

We have an automaticatic vacuum(autovacuum) that runs every minute. This can be confirmed with:

```sql
> select setting, unit from pg_settings where name = 'autovacuum_naptime';
60 s
```

---

> i am wondering does deleting rows periodically frees up this space or does the deleted rows count too?

@Nisthar So yes, but not immediately, as mentioned above. You can run `VACUUM` manually for freeing up space immediately,  though it's recommended that you let `autovacuum` do the job.

**Reply by Nisthar** [reply]
Is there a way to check the current size usage from the dashboard? @steve-chavez

**Reply by silentworks** [reply]
Thanks for this bit of info @steve-chavez.
